File: ark.py
#!/usr/bin/env python3

import argparse
import json
import requests

def main():
    try:
        parser = argparse.ArgumentParser(description='DoThings')
        parser.add_argument("--ipaddress", required=True)
        parser.add_argument("--port", required=False, default=27015)
        args = parser.parse_args()
        request = requests.get("http://arkservers.net/api/query/" + args.ipaddress + ":" + args.port)
        response = request.content
        if response == b"null":
            print("Server is Offline")
            exit(1)
        else:
            jsonData = json.loads(response)
            print(jsonData["info"]["HostName"] + ": " + str(jsonData["info"]["Players"]) + "/" + str(jsonData["info"]["MaxPlayers"]))
    except Exception as e:
        print(e)

main()

File: ark.php
<?php
if(!array_key_exists(1, $argv)){
  echo "Ip address is required";
  exit(1);
}

$ipAddress = $argv[1];

if(!array_key_exists(2, $argv)){
  $port = 27015;
}else{
  $port = $argv[2];
}

$response = file_get_contents("http://arkservers.net/api/query/{$ipAddress}:{$port}");
if($response === "null"){
  echo "Server is Offline\n";
  exit(1);
}

$serverData = json_decode($response);
echo "{$serverData->info->HostName}: {$serverData->info->Players}/{$serverData->info->MaxPlayers}\n";
exit(0);
